ols/utils/config.py

```diff
@@ -23,130 +23,36 @@ def load_empty_config() -> None:
-        redis_config.max_memory_policy = constants.REDIS_CACHE_MAX_MEMORY_POLICY
-    elif cache_config.type == constants.IN_MEMORY_CACHE:
-        memory_config = config_model.MemoryConfig({})
-        cache_config.memory = memory_config
-        memory_config.max_entries = int(
-            os.getenv("MEMORY_CACHE_MAX_ENTRIES", constants.IN_MEMORY_CACHE_MAX_ENTRIES)
-        )
-    else:
-        raise Exception(f"Invalid cache type: {cache_config.type}")
-    conversation_cache = CacheFactory.conversation_cache(cache_config)
-
-    llm_config = config_model.LLMConfig()
-    if os.getenv("BAM_API_KEY", None) is not None:
-        bam_provider = config_model.ProviderConfig()
-        bam_provider.name = constants.PROVIDER_BAM
-        bam_provider.credentials = os.getenv("BAM_API_KEY", None)
-        bam_provider.url = os.getenv("BAM_API_URL", "https://bam-api.res.ibm.com")
-
-        for model in [
-            constants.GRANITE_13B_CHAT_V1,
-            constants.GRANITE_13B_CHAT_V2,
-            constants.GRANITE_20B_CODE_INSTRUCT_V1,
-        ]:
-            m = config_model.ModelConfig()
-            m.name = model
-            bam_provider.models[m.name] = m
-        llm_config.providers[bam_provider.name] = bam_provider
-
-    if os.getenv("OPENAI_API_KEY", None) is not None:
-        oai_provider = config_model.ProviderConfig()
-        oai_provider.name = constants.PROVIDER_OPENAI
-        oai_provider.credentials = os.getenv("OPENAI_API_KEY", None)
-        oai_provider.url = os.getenv("OPENAI_API_URL", "https://api.openai.com/v1")
-
-        for model in [
-            constants.GPT35_TURBO_1106,
-            constants.GPT35_TURBO,
-        ]:
-            m = config_model.ModelConfig()
-            m.name = model
-            oai_provider.models[m.name] = m
-        llm_config.providers[oai_provider.name] = oai_provider
+    return c
 
 
-# def load_config(config_file):
-#     try:
-#         with open(config_file, "r") as f:
-#             data = yaml.safe_load(f)
-#             c=Config(data)
-#     except Exception as e:
-#         print(f"Failed to load config file {config_file}: {str(e)}")
-#         print(traceback.format_exc())
-#         exit(1)
-#     try:
-#         c.validate()
-#     except Exception as e:
-#         print(f"Failed to validate config file {config_file}: {str(e)}")
-#         print(traceback.format_exc())
-#         exit(1)
-#     global config
-#     global default_logger
-#     global feedback_logger
-#     global conversation_cache
-#     config = c
-#     default_logger = Logger(logger_name="default", logfile=c.ols_config.logger_config.default_filename,log_level=c.ols_config.logger_config.default_level).logger
-#     feedback_logger = Logger(logger_name="feedback", logfile=c.ols_config.logger_config.feedback_filename,log_level=c.ols_config.logger_config.feedback_level).logger
+def load_config(config_file):
+    """Load configuration from a file."""
+    global config
```

Comment:
dtto, it uses global var and returns it at the same time, IMHO that's confusing

Reply:
on this point i agree.  Can probably just call it "init_config" and have it initialize the global var w/o returning it.

Reply:
What about wrapping all the config objects to `OLSConfig`? The `init_config` can then return the config as a whole in the `main` (at the app entrypoint). So we would have something like this there
```python
config = init_config()
```
Then you can access this global everywhere in the app like
```python
from ols.main import config

redis_host = config.ols_config.conversation_cache.redis.url
```

Reply:
Agree 100%. I'm hoping to get rid of global ols_config and llm_config, just in a different PR, or this one will never merge :)